### ISSUE TITLE
RFC: Approach for paginated APIs on canisters

### ISSUE THREAD

**Issue by saikatdas0790** [go-bazzinga/hot-or-not-backend-canister#273]
For all paginated APIs on our platform I think we should have a uniform assumption. They have to be cursor based. You pass in the ID of the last item you've received as the first argument and number of items you want as the 2nd argument and you're returned a list/array/vec of the expected items. Something like this:

```rust
get_paginated_items(id_of_last_item_received: u64, number_of_items_to_be_returned: u64, filter_criteria_if_any: Option<Filter>) -> Vec<Item> {}
```

In the implementation, we can then rely on iterator to iterate over the items asked for, apply any filters that are specified and then return the number of items asked for with an upper bound or iterated till the end of the list, whichever is earlier and then just return a cloned list of the items generated from the iteration.

Thoughts?

Should permanently solve issues such as #268 and is also an efficient iteration approach that solves any filtering that we'll want to apply

**Comment by ravibazz** [comment]
Yes makes sense, just one corner case instead of last id we can keep offset because not every api will return items sorted by id in that case it would create confusion.
